.github/workflows/auto_branching.yml

```
@@ -0,0 +1,294 @@
+
+          if [[ $PROTECTION_RESPONSE -eq 200 ]]; then
+            echo "Branch protection successfully updated."
+            echo "protection-outcome=success" >> "$GITHUB_OUTPUT"
+          else
+            echo "Failed to update branch protection. HTTP status code: $PROTECTION_RESPONSE"
+            echo "protection-outcome=failure" >> "$GITHUB_OUTPUT"
+            exit 1
+          fi
+
+  auto-branching-master:
+    name: master - raise PR with changes
+    runs-on: ubuntu-latest
+    needs: check-group-membership
+    if: ${{ needs.check-group-membership.outputs.member == 'true' }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Update target branch label in dependabot yml file
+        id: update-dependabot
+        run: |
+          # Read the dependabot.yml file
+          FILE_PATH="./.github/dependabot.yml"
+          TARGET_BRANCH="${{ github.event.inputs.target_branch }}"
+          # Append the target branch label to the labels node
+          awk -v target="'$TARGET_BRANCH'" '/^ *labels:/ {$0 = $0 "\n      - " target} 1' "$FILE_PATH" > temp.yml && mv temp.yml "$FILE_PATH"
+
+      - name: Remove lines with @pytest.mark.stream
+        id: remove-mark-stream
+        run: |
+          # Loop through files in the folder
+          grep -rl "tests/foreman" -e '@pytest\.mark\.stream' | while IFS= read -r file; do
+           awk '!/@pytest\.mark\.stream/' "$file" > temp && mv temp "$file"
+          done
+
+      - name: Update the Constants in __init__.py file
+        run: |
+          version="${{ github.event.inputs.target_branch }}"
+          ga_version="${{ github.event.inputs.ga_version }}"
+          old_stream_version="${version%.z}"
+          new_stream_version="${{ github.event.inputs.stream_version }}"
+          non_ga_versions="['$old_stream_version', '$new_stream_version']"
+          FILE_PATH="./robottelo/constants/__init__.py"
+          # update the version
+          sed -i.bak "s/SATELLITE_VERSION = \"$old_stream_version\"/SATELLITE_VERSION = \"$new_stream_version\"/" "$FILE_PATH"
+          sed -i.bak "s/  SATELLITE_VERSION: \"$old_stream_version\"/  SATELLITE_VERSION: \"$new_stream_version\"/" ./conf/robottelo.yaml.template
+          sed -i.bak "s/SAT_NON_GA_VERSIONS = \[.*\]/SAT_NON_GA_VERSIONS = $non_ga_versions/" "$FILE_PATH"
+          rm "$FILE_PATH.bak" "./conf/robottelo.yaml.template.bak"
+
+      - name: git status
+        run: git status
+
+      - name: git diff
+        run: git diff
+
+      - name: Commit changes
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+          branch_name="auto-branching-${{ github.event.inputs.target_branch }}-$(date '+%s')"
+          git checkout -b "$branch_name"
+          git add setup.py ./tests/foreman ./robottelo/* ./requirements.txt ./.github/* ./conf/robottelo.yaml.template
```

why not add everything changed?
```suggestion
          git add .
```

---

it's often safer to use the explicit approach to avoid accidentally including unintended changes.

---

In any case, it'll open up a PR for us to review and merge, and if we encounter any accidentally or unintended changes in it, we can manually mitigate them, right?

---

I don't think so it is right approach `git add .`. I personally never do that on my local machine even. @jyejare @JacobCallahan do you feel the same ?

---

`git add -u` only adds file those which are already part of vc and not those are new. So in any case if we add a new file now or later then it would be missed.

So its better to be explicit here.